src/OpenTelemetry.Api/Trace/TelemetrySpan.cs

```diff
@@ -380,5 +383,21 @@ protected virtual void Dispose(bool disposing)
         {
             this.Activity?.Dispose();
         }
+
+        private void SetAttributeInternal(string key, object value)
+        {
+            if (this.IsRecording)
+            {
+                this.Activity?.SetTag(key, value);
+            }
+        }
+
+        private void AddEventInternal(string name, DateTimeOffset timestamp = default, ActivityTagsCollection tags = null)
+        {
+            if (this.IsRecording)
+            {
+                this.Activity?.AddEvent(new ActivityEvent(name, timestamp, tags));
```

Comment:
null check not needed here as isrecording already does it.

Reply:
ok. i will remove in both places.